src/app/pages/signature-request/components/message-box.tsx

```diff
@@ -8,16 +8,15 @@ import { isStructuredMessage, SignatureMessage } from './sign-action';
 
 export function MessageBox(props: SignatureMessage): JSX.Element | null {
   const { message, messageType } = props;
-
   const [hash, setHash] = useState<string | undefined>();
-  const [displayMessage, setDisplayMessage] = useState<string | undefined>();
+  const [displayMessage, setDisplayMessage] = useState<string[] | undefined>();
   const [clarityValueMessage, setClarityValueMessage] = useState<ClarityValue | undefined>();
 
   useEffect(() => {
     if (isStructuredMessage(messageType)) {
       setClarityValueMessage(deserializeCV(Buffer.from(message, 'hex')));
     } else {
-      setDisplayMessage(message);
+      setDisplayMessage(message.split(/\r?\n/));
```

Comment:
This is really just a thought, no need to change, but in some ways, I think we shouldn't need to change the _state_ of the message just to handle some formatting requirement.

Maybe just a `formatDisplayMessage(message)` could've handled the split, without needing to adjust the component's state? Then we'd be handling formatting concerns in the jsx output part of the code, rather than the state-setting part.

Reply:
Ok, thanks for the suggestion, I can try to change it with another PR at some point.